projects/packages/connection/src/sso/class-sso.php

```diff
@@ -0,0 +1,1280 @@
+<?php
+/**
+ * SSO feature. Entry point.
+ *
+ * @package automattic/jetpack-connection
+ */
+
+namespace Automattic\Jetpack\Connection;
+
+use Automattic\Jetpack\Connection\SSO\Force_2FA;
+use Automattic\Jetpack\Connection\SSO\Helpers;
+use Automattic\Jetpack\Connection\SSO\Notices;
+use Automattic\Jetpack\Connection\SSO\User_Admin;
+use Automattic\Jetpack\Roles;
+use Automattic\Jetpack\Status;
+use Automattic\Jetpack\Status\Host;
+use Automattic\Jetpack\Tracking;
+use Jetpack;
+use Jetpack_IXR_Client;
+use WP_Error;
+use WP_User;
+use WP_User_Query;
+
+/**
+ * SSO feature main class.
+ */
+class SSO {
+	/**
+	 * WordPress.com User information.
+	 *
+	 * @var false|object
+	 */
+	private $user_data;
+
+	/**
+	 * Automattic\Jetpack\Connection\SSO instance.
+	 *
+	 * @var \Automattic\Jetpack\Connection\SSO
+	 */
+	public static $instance = null;
+
+	/**
+	 * Automattic\Jetpack\Connection\SSO constructor.
+	 */
+	private function __construct() {
+
+		self::$instance = $this;
+
+		/*
+		 * This feature currently relies on the Jetpack plugin.
+		 * Bail if Jetpack isn't installed.
+		 */
+		if ( ! class_exists( 'Jetpack' ) ) {
+			return;
+		}
+
+		add_action( 'admin_init', array( $this, 'maybe_authorize_user_after_sso' ), 1 );
+		add_action( 'admin_init', array( $this, 'register_settings' ) );
+		add_action( 'login_init', array( $this, 'login_init' ) );
+		add_filter( 'jetpack_xmlrpc_methods', array( $this, 'xmlrpc_methods' ) );
+		add_action( 'init', array( $this, 'maybe_logout_user' ), 5 );
+		add_action( 'login_form_logout', array( $this, 'store_wpcom_profile_cookies_on_logout' ) );
+		add_action( 'jetpack_unlinked_user', array( Helpers::class, 'delete_connection_for_user' ) );
+
+		add_action( 'jetpack_site_before_disconnected', array( static::class, 'disconnect' ) );
+		add_action( 'wp_login', array( static::class, 'clear_cookies_after_login' ) );
+
+		// Adding this action so that on login_init, the action won't be sanitized out of the $action global.
+		add_action( 'login_form_jetpack-sso', '__return_true' );
+
+		add_filter( 'wp_login_errors', array( $this, 'sso_reminder_logout_wpcom' ) );
+
+		/**
+		 * Filter to include Force 2FA feature.
+		 *
+		 * By default, `manage_options` users are forced when enable. The capability can be modified
+		 * with the `jetpack_force_2fa_cap` filter.
+		 *
+		 * To enable the feature, add the following code:
+		 * add_filter( 'jetpack_force_2fa', '__return_true' );
+		 *
+		 * @param bool $force_2fa Whether to force 2FA or not.
+		 *
+		 * @todo Provide a UI to enable/disable the feature.
+		 *
+		 * @since jetpack-12.7
+		 * @module SSO
+		 * @return bool
+		 */
+		if (
+			! class_exists( 'Jetpack_Force_2FA' )
```

Comment:
Is this the right class? Or should we be referencing the new Class `Force_2FA` from the connection package?

Reply:
Good catch! I fixed that in 632d13dcd35773053498660b18a5c6d9e5d0bd64

Reply:
Looks like `class_exists( 'Force_2FA' )` is always false because of the namespace mixup.
You probably need to use the full namespace path:
```php
class_exists( 'Automattic\Jetpack\Connection\SSO\Force_2FA' )
```

Reply:
Ah yes, good point, fixed in fa01824152a242e5c11b88f6f9c104139ce03348